dcos_test_utils/etcd.py

```diff
@@ -0,0 +1,45 @@
+import os
+import subprocess
+
+from typing import List
+
+ETCDCTL_PATH = "/opt/mesosphere/bin/etcdctl"
+ETCD_ENDPOINTS = "127.0.0.1:2379"
+
+CA_CERT = "/run/dcos/pki/CA/ca-bundle.crt"
+
+
+def is_enterprise():
+    return os.getenv('DCOS_ENTERPRISE', 'false').lower() == 'true'
+
+
+class EtcdCtl():
+    """ wraps etcdctl around related configurations
+    """
+
+    def __init__(self, cert_type="root") -> None:
+        self._base_args = self._get_base_args(cert_type)
+
+    def _get_base_args(self, cert_type) -> List[str]:
+        args = ["sudo", ETCDCTL_PATH]
+        if is_enterprise():
+            args += ["--endpoints=https://{}".format(ETCD_ENDPOINTS)]
+            args += [
+                "--cert=/run/dcos/pki/tls/certs/etcd-client-{}.crt".format(cert_type),
+                "--key=/run/dcos/pki/tls/private/etcd-client-{}.key".format(cert_type),
+                "--cacert={}".format(CA_CERT),
+            ]
+        else:
+            args += ["--endpoints=http://{}".format(ETCD_ENDPOINTS)]
+
+        return args
+
+    def run(self, cmd: List[str], check: bool = True,
+            env: dict = {}) -> subprocess.CompletedProcess:
+        process = subprocess.run(
+            self._base_args.copy() + cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            env=env,
+            check=check)
+        return process
```

Comment:
Not blocking, but since you are using `subprocess.PIPE` it's expected that whoever is receiving this `subprocess.CompletedProcess` object also consumes the pipe buffers. I feel that this might introduce future bugs (eg. someone reading only one of the two pipes and eventually reaching a deadlock on the other pipe if the OS buffer fills-up). Pretty rare case, but still possible 😄 

I could recommend performing a `.communicate` after launching the process and returning a triplet of `(return code, stdout, stderr)`

Reply:
For instance, consider someone reading only the `.stdout` from the subprocess and you hit the warning case mentioned here : https://docs.python.org/3/library/subprocess.html#subprocess.Popen.stderr

Reply:
I did some research and `subprocess.run` already calls `.communicate` internally:

https://github.com/python/cpython/blob/3.8/Lib/subprocess.py#L491

:)

Reply:
Oh, nice so the `CompletedProcess` already has the buffers populated. Cool then! 👍